File: data/src/main/java/com/tonitealive/app/data/net/ApiAuthInterceptor.java
package com.tonitealive.app.data.net;

import com.google.common.base.Optional;
import com.tonitealive.app.data.TokenStore;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

class ApiAuthInterceptor implements Interceptor {

    private final TokenStore tokenStore;

    public ApiAuthInterceptor(TokenStore tokenStore) {
        this.tokenStore = tokenStore;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        Optional<String> token = tokenStore.getAuthToken();
        if (token.isPresent()) {
            request = request.newBuilder()
                        .addHeader("Authorization", "Bearer " + token.get())
                        .build();
        }
        return chain.proceed(request);
    }
}

File: data/src/main/kotlin/com/tonitealive/app/data/net/ApiAuthInterceptor.kt
package com.tonitealive.app.data.net

import com.tonitealive.app.data.TokenStore
import okhttp3.Interceptor
import okhttp3.Response


internal class ApiAuthInterceptor(private val tokenStore: TokenStore) : Interceptor {

    override fun intercept(chain: Interceptor.Chain?): Response {
        var request = chain!!.request()
        val authToken = tokenStore.authToken
        if (authToken != null) {
            request = request.newBuilder()
                    .addHeader("Authorization", "Bearer ${authToken.accessToken}")
                    .build()
        }
        return chain.proceed(request)
    }

}